app/controllers/api/pages_controller.rb

```diff
@@ -37,6 +38,20 @@ def featured
     render :index, format: :json
   end
 
+  def actions
+    return head :forbidden if @page.secure?
+    query = if @page.default_hidden?
+              published_status = Action.publish_statuses['published']
+              "page_id = '#{@page.id}' AND publish_status = '#{published_status}'"
+            else
+              hidden_status = Action.publish_statuses['hidden']
+              "page_id = '#{@page.id}' AND publish_status != '#{hidden_status}'"
+            end
+    page_number = { page_number: params[:page_number], per_page: params[:per_page] }
+    hashes, headers, _paginator = ActionReader.new(query).run(**page_number)
+    render json: { actions: hashes, headers: headers }
+  end
```

Comment:
My priority for the ActionReader class was that it be reusable in the two places I'm using it. It already is a service class that has another service class that only gets called by it, three levels seems like a lot...